test/Microsoft.AspNetCore.Mvc.Core.Test/Formatters/FormatterMappingsTest.cs

```diff
@@ -49,14 +49,16 @@ public void FormatterMappings_Invalid_Period()
             // Arrange
             var options = new FormatterMappings();
             var format = ".";
-            var expected = string.Format(@"The format provided is invalid '{0}'. A format must be a non-empty file-" +
-                "extension, optionally prefixed with a '.' character.", format);
+            var expected = $@"The format provided is invalid '{format}'. A format must be a non-empty file-" +
+                "extension, optionally prefixed with a '.' character.";
 
             // Act and assert
-            var exception = Assert.Throws<ArgumentException>(() => options.SetMediaTypeMappingForFormat(
-                format,
-                MediaTypeHeaderValue.Parse("application/xml")));
-            Assert.Equal(expected, exception.Message);
+            var exception = ExceptionAssert.ThrowsArgument(
+                () => options.SetMediaTypeMappingForFormat(
+                    format,
+                    MediaTypeHeaderValue.Parse("application/xml")),
+                null,
```

Comment:
This is a bug in the product code `ValidateContentType()` and `RemovePeriodIfPresent()` should not be `throw`ing `ArgumentException`s w/o parameter names. Fix or file a follow-up.